src/ggml.c

```diff
@@ -12166,15 +12167,16 @@ static void ggml_compute_forward_upscale_f32(
     // TODO: optimize
 
     for (int i03 = 0; i03 < ne03; i03++) {
-        for (int i02 = ith; i02 < ne02; i02++) {
-            for (int m = 0; m < dst->ne[1]; m++) {
-                int i01 = m / scale_factor;
-                for (int n = 0; n < dst->ne[0]; n++) {
-                    int i00 = n / scale_factor;
-
-                    const float * x = (float *)((char *) src0->data + i00 * nb00 +i01 * nb01 + i02 * nb02 + i03 * nb03);
-
-                    float * y = (float *)((char *) dst->data + n * dst->nb[0] + m * dst->nb[1] + i02 * dst->nb[2] + i03 * dst->nb[3]);
+        const int i3 = i03;
+        for (int i02 = ith; i02 < ne02; i02 += nth) {
```

Comment:
Used in the YOLO example, but don't think we ever hit it since it would happen for 3D or 4D tensors